meson.build

```
@@ -72,24 +72,12 @@ if not get_option('source-only')
 
     default_fallback_options = ['warning_level=0', 'werror=false']
 
-    # Lua has no official .pc file
-    # so distros come up with their own names
-    lua_names = [
-        'lua5.4', # Debian
-        'lua-5.4', # FreeBSD
-        'lua',    # Fedora
-    ]
-
-    foreach lua : lua_names
-        last_lua = (lua == lua_names[-1])
-        lua_dep = dependency(lua, fallback: last_lua ? ['lua', 'lua_dep'] : [], required : last_lua,
-            version: '>= 5.4',
-            default_options: default_fallback_options + ['default_library=static', 'line_editing=false', 'interpreter=false']
-        )
-        if lua_dep.found()
-            break
-        endif
-    endforeach
+    # Lua has no official .pc file so distros come up with their own names
+    # lua5.4 -> Debian | lua-5.4 -> FreeBSD | lua -> Fedora
+    lua_dep = dependency('lua5.4', 'lua-5.4', 'lua', fallback: ['lua', 'lua_dep'],
```

```suggestion
    lua_dep = dependency(lua_names, fallback: ['lua', 'lua_dep'],
```

Using the original array with inline comments feels a bit cleaner.

---

> Note that this was explicitly avoided during the initial implementation because it requires bumping the minimum version of Meson.

I see (didn't noticed the meson warnings telling me it is a feature of 0.60), the issue is it doesn't obeys the forcefallback which we need for proper CI builds, are there any platforms where getting a newer meson to build would be impossible?

---

> Using the original array with inline comments feels a bit cleaner.

I thought the same initially but didn't tested to see if that was supported

---

A backwards-compatible solution might be:

```meson
required : last_lua
```

This marks the dependency as required only after several others have tried and failed. But forcefallback only takes effect for cases where subprojects are considered, which means:
- dependencies that are required
- dependencies that use 0.56.0+ `allow_fallback: true`

So, change this to:

```meson
required: last_lua or get_option('wrap_mode') == 'forcefallback'
```

---

> are there any platforms where getting a newer meson to build would be impossible?

Depends on what "getting a newer Meson" means. Some people want to be compatible with whichever version of Meson is available in the distro repositories. That means Ubuntu Focal (20.04) with 0.53.0 would be a cutoff. Clearly you don't both care about the distro Meson and care about supporting Ubuntu Bionic (18.04) with its Meson 0.45.1.

Some people are perfectly happy expecting users to `pip install -U meson` if they want to build the project. In that case, the current requirement on git master is a version of Meson that requires python 3.5, and this PR bumps that to a version that requires python 3.6.

The latest version of Meson, meanwhile, requires python 3.7 and has done so ever since python upstream marked 3.6 as EOL. Versions of Ubuntu older than Bionic have python older than 3.6 and could have pip installed Meson before this PR, but cannot do so after this PR.

What scenarios are you interested in supporting?